maths/perfect_cube.py

```diff
@@ -11,6 +11,39 @@ def perfect_cube(n: int) -> bool:
     return (val * val * val) == n
 
 
+def perfect_cube_binary_search(n: int) -> bool:
+    """
+    Check if a number is a perfect cube or not using binary search.
+    Time complexity : O(Log(n))
+    Space complexity: O(1)
+
+    >>> perfect_cube_binary_search(27)
+    True
+    >>> perfect_cube_binary_search(64)
+    True
+    >>> perfect_cube_binary_search(4)
+    False
+    >>> perfect_cube_binary_search("a")
+    Traceback (most recent call last):
+        ...
+    TypeError: '<=' not supported between instances of 'int' and 'str'
```

Comment:
Please add your own TypeError check for invalid inputs like this rather than relying on the built-in error message for `<=`